File: impls/myrust/step0_repl.rs
use std::io::{self, Write};

fn read(input: &String) -> &String {
    input
}

fn eval(input: &String) -> &String {
    input
}

fn print(input: &String) -> &String {
    input
}

fn rep(input: &String) -> &String {
    print(eval(read(input)))
}

fn main() {
    print!("user> ");
    io::stdout().flush().expect("couldn't flush stdout");
    let mut input = String::new();

    let mut bytes = io::stdin()
        .read_line(&mut input)
        .expect("could not read line");

    while bytes != 0 {
        println!("{}", rep(&input).trim());

        print!("user> ");
        io::stdout().flush().expect("couldn't flush stdout");
        input = String::new();
        bytes = io::stdin()
            .read_line(&mut input)
            .expect("could not read line");
    }
}

File: impls/synpy/step0_repl.py
def READ(string):
    return string


def EVAL(ast, env):
    return ast


def PRINT(exp):
    return exp


def rep(string):
    return PRINT(EVAL(READ(string), ""))


if __name__ == "__main__":
    while True:
        print(rep(input("user> ")))
